src/plugins/links/public/components/dashboard_link/dashboard_link_component.tsx

```diff
@@ -116,15 +84,10 @@ export const DashboardLinkComponent = ({
 
     const params: DashboardLocatorParams = {
       dashboardId: link.destination,
-      ...getDashboardLocatorParamsFromEmbeddable(
-        linksEmbeddable as Partial<
-          PublishesUnifiedSearch & HasParentApi<Partial<PublishesUnifiedSearch>>
-        >,
-        linkOptions
-      ),
+      ...getDashboardLocatorParamsFromEmbeddable(api, linkOptions),
     };
 
-    const locator = dashboardContainer.locator;
+    const locator = api.parentApi.locator;
```

Comment:
It seems like the locator is only getting updated on first load, so filters + queries are only carried over if they were there when the dashboard was loaded - new filters/queries are not carried over:


https://github.com/elastic/kibana/assets/8698078/38c78b0d-232b-4870-a008-08b53edea7c0

Reply:
Should be fixed with the latest changes now. Thanks for catching this.